File: model/monetization/ProductPlanPriceModel.java
package com.digitinary.dgate.model.monetization;

import com.digitinary.dgate.model.TimeUnit;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import java.math.BigDecimal;
import java.util.Set;

/**
 * This class represents a product plan price. A product will have a list of prices for each different plan.
 *
 * @author Salah Abu Msameh
 * @since 29/09/2024
 */
@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ProductPlanPriceModel {

    private Long id;
    private ProductSimpleViewModel product;
    private PlanSimpleViewModel plan;
    private String planName;
    private Set<SubscriptionPeriodOption> subscriptionPeriodOptions;
    private Set<SubscriptionRenewalType> renewalTypeOptions;
    private ApiPricingType priceType;
    private BigDecimal monthlyPrice;
    private BigDecimal yearlyPrice;
    private BigDecimal lifetimePrice;
    private Long apiCallsQuota;
    private TimeUnit timeUnit;
    private Set<String> missingProductNames;
}

File: model/monetization/ProductPlanPriceModel.py
class represents:
    def __init__(self):
        self.id = None  # Long
        self.product = None  # ProductSimpleViewModel  # This might be an enum
        self.plan = None  # PlanSimpleViewModel  # This might be an enum
        self.planName = None  # String
        self.subscriptionPeriodOptions = None  # Set<SubscriptionPeriodOption>
        self.renewalTypeOptions = None  # Set<SubscriptionRenewalType>
        self.priceType = None  # ApiPricingType  # This might be an enum
        self.monthlyPrice = None  # BigDecimal  # This might be an enum
        self.yearlyPrice = None  # BigDecimal  # This might be an enum
        self.lifetimePrice = None  # BigDecimal  # This might be an enum
        self.apiCallsQuota = None  # Long
        self.timeUnit = None  # TimeUnit  # This might be an enum
        self.missingProductNames = None  # Set<String>